docs/specs/alcproxies.md

```diff
@@ -0,0 +1,251 @@
+Assembly aa = alc.LoadFromAssemblyPath(assemblyPath); //Load the assembly into the ALC
+Type classType = aa.GetType("AssemblyName.DesiredType");
+
+// For instantiating below, you can also use Activator.GetInstance(classType) if you know there is a default constructor that takes no arguments
+object obj = (classType.GetConstructor(Type.EmptyTypes).Invoke(new object[] { }));
+
+obj.GetType().GetMethod("SomeMethod").Invoke(obj, null);
+```
+The problem here is that due to Type Isolation of ALCs, you cannot directly cast the instantiated object to its original type or an inherited type, as the CLR believes them to be different types from what's loaded from the ALC. 
+
+There are two ways to use the instance from the ALC:
+* Use reflection to invoke calls of the objects type, gotten directly using object.GetType()
+* Use a shared assembly between the user and target assembly that holds a class/interface that the type you want inherits from, and using said shared type instead of the type you want directly (See [this example](https://github.com/dotnet/samples/tree/master/core/extensions/AppWithPlugin ) to see it working).
+* Use `DispatchProxy` to create a proxy within the user ALC, pointing to an object in the target ALC. This also requires a shared assembly.
+
+Anything requiring a shared assembly is inadequate for what we're trying to do, as one of the main benefits of AppDomains having totally separate types was with version compatibility. In .NET Framework, different versions of assemblies could be loaded into separate AppDomains, and communication could occur between objects as if they were one shared type. For example, if a user has a program that uses dependency A, and wants to load a plugin that has classes that inherit from older versions of dependency A, AppDomains would still be able to send and receive information between the plugin and the main program. If we enforce shared assemblies with this API, we lose the ability to allow for things like this.
+
+## Proposed Solution
+
+### Goals
+This proposal is for an API that can effectively recreate the object creation capabilities in AppDomain. The goals of this design are:
+* To allow for people are able to create isolated objects without the direct use of reflection. 
+* To preserve AppDomains ease of use when creating isolated objects.
+* To continue support for ALC unloads by destroying instantiated object when their ALC is unloaded, as well as other ALC features.
+* To allow for plugin compatibility to work even with minor versioning changes being made to the application.
+* Attempt to design around allowing inter-process communication between the user's process and the process where the object instance will be made.
+
+One thing that's especially interesting is use of this design to not only allow for communication between ALCs, but expanding the design to work for inter-process communication, which is why the 5th bullet point is there.
+
+### Design
+```C#
```

Comment:
Thank you for leading with code samples, it makese understanding the design much simplier/easy.  An area I would like us to explore deeper regarding our interface approach is in what assembly this interface resides.  At runtime we need to load something, and often it may be part of a larger assembly.  For me the end goal would be that the developer does not have to do anything unnatural to use the Proxied object.